astro/src/content/quickstarts/quickstart-javascript-nextjs-web.mdx

```diff
@@ -0,0 +1,277 @@
+* Your example username is `richard@example.com` and the password is `password`.
+* Your admin username is `admin@example.com` and the password is `password`.
+* The base URL of FusionAuth `http://localhost:9011/`.
+
+You can log into the [FusionAuth admin UI](http://localhost:9011/admin) and look around if you want, but with Docker/Kickstart you don’t need to.
+
+### Full Example
+
+Make sure once again that you are in the directory `complete-application/`.
+
+Copy the `.env.example` to `.env.local`
+
+```shell
+cp .env.example .env.local
+```
+
+The environment variables will match what is in the `/kickstart/kickstart.json` file that is used when first starting FusionAuth. If you have changed anything, like creating a new Application you will need to update these environment variables accordingly.
+
+`.env.local`
+<RemoteCode url={frontmatter.coderoot + "/complete-application/.env.example"}
+  lang="ts" />
+
+Install required packages
+
+```shell
+npm install
+```
+
+### Walkthrough Example
+
+In this section, you'll set up a basic Next.js application with two pages. 
+
+1. Homepage
+1. Account - protected
+
+#### Setup Your Environment
+
+Create a new application using the `npx`.
+
+```shell
+npx create-next-app@latest changebank
+```
+
+Accept all of the default prompts like below
+
+![npx installation defaults](/img/articles/quickstart-javascript-nextjs-web/npx-create-next-app.png)
+
+Make sure you are in your new directory `changebank`
+
+```shell
+cd changebank
+```
+
+Install [NextAuth.js](https://next-auth.js.org/getting-started/example), like any tutorial I will show you what this magic does later.
+
+```shell
+npm install next-auth
+```
+
+Copy environment variables from our complete application example.
+
+```shell
+cp ../complete-application/.env.example .env.local
+```
+
+Also copy an image file into a new directory within `public` called `img`.
+
+```shell
+mkdir ./public/img && cp ../complete-application/public/img/money.jpg ./public/img/money.jpg
+```
+
+As you will be recreating all of the files in our app directory, please delete all files within `/src/app`.
+
+```shell
+rm -rf ./src/app && mkdir ./src/app
+```
+
+#### Authentication API
+
+Next.js 13.2 introduced <a href="https://nextjs.org/docs/app/building-your-application/routing/router-handlers" target="_blank">Route Handlers</a>, which are the preferred way to handle REST-like requests. In the `Changebank` application you can configure NextAuth.js FusionAuth's provider in a new route handler by creating a file within `src/app/api/auth/[...nextauth]/route.ts`.
+
+On first load of Next.js this file will make sure that you have all of the correct environment variables. The variables are then exported in an object called `authOptions` which can be imported on the server when you need to get our session using `getServerSession`.
+
+The `FusionAuthProvider` is then provided to `NextAuth` as a provider for any `GET` or `POST` commands that are sent to the `/api/auth/*` route.
```

Comment:
Since I just removed `src/app` per your instructions, I don't know how to create this file. Do I need to create some directories?

Reply:
I am not sure I follow, you should just be creating the file `src/app/api/auth/[...nextauth]/route.ts`

Is the issue that you would like to create all these directories separately?

I usually create any file like this...
<img width="369" alt="image" src="https://github.com/FusionAuth/fusionauth-site/assets/139158618/bca78940-4ad9-4240-bf4b-c82672eb3540">